metadata.yaml

```diff
@@ -100,6 +100,14 @@ requires:
       Receive a CA cert for traefik to trust.
       This relation can be used with a local CA to obtain the CA cert that was used to sign proxied
       endpoints.
+      This is meant for applications that use the certificate-transfer-interface v0.
+  receive-ca-cert-v1:
+    interface: certificate_transfer
+    description: |
+      Receive a CA certs for traefik to trust.
+      This relation can be used with a local CA to obtain the CA certs that was used to sign proxied
+      endpoints.
+      This is meant for applications that use the certificate-transfer-interface v1.
```

Comment:
My uninformed first impression is that I dislike adding a new relation to handle a different version of the same interface.  This feels like a change that should be transparent to the user if at all possible.  Ideally, the existing `receive-ca-cert` relation should just work for both v0 and v1.  Is that possible?

Reply:
If the interface is the same, we shouldn't use a new relation. The promise is supposed to be that any provider of an interface works with any consumer.

Reply:
V0 and V1 are not compatible. There was breaking change in the interface too not only the library.
Here you can see examples of the data expected in [V0](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/certificate_transfer/v0/README.md#example) and the data expected in [V1](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/certificate_transfer/v1/README.md#example).

Isn't it the same with Ingress V1 and V2?

Reply:
It's the same for ingress but handled differently not to require a new relation, traefik [only has one relation for ingress](https://github.com/canonical/traefik-k8s-operator/blob/main/metadata.yaml#L46) and we do our checks internally within the charm code to [validate if client is talking v0/v1](https://github.com/canonical/traefik-k8s-operator/blob/main/src/charm.py#L890)

Reply:
I see your point thank you.
I'll update the PR.

Reply:
I updated the PR to follow the example of Ingress.
Could you please take a look @IbraAoad ?